### PR TITLE
Web storage docs

### PR DESCRIPTION
**Pull request by gosteenBASIS**
Web storage docs page completed to the best of my knowledge, with a couple caveats: 
- Not sure where it should go in the project structure. Currently placed under "Architecture."
- Have not actually tested the sample code, although it appears correct based on the Javadocs
- Note about the default 30-day expiration of the CookieStorage is based on comments in the code; have not verified otherwise.
- According to the Javadocs, it looks like you can pass attributes to the constructors for SessionStorage and LocalStorage, but based on some brief research it doesn't look like they can actually take attributes like cookies do. I have omitted any mention of attributes in the sample code and description for those sections.

### CODE REVIEW

**Comment by MatthewHawkins** [comment]
Hey @gosteenBASIS, thanks for finishing this up, looks good!

- I asked Hyyan where he would like the article to be put within the hierarchy, and will do that once he answers before I ask for his review
- I tested and ran into a small issue that I'm hammering out with Hyyan. We can talk about testing code for the next few articles at some point next week

Thanks again, and once Hyyan and I tie up things, we'll get this merged!

**Review by MatthewHawkins — APPROVED** [review]

**Review by hyyan — CHANGES_REQUESTED** [review]

File: docs/advanced_topics/web-storage.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+### Advantages:
+- Can store data across different domains
+- Support expiration dates
+
+### Disadvantages:
+- Small storage size (about 4KB)
+- Sent with every HTTP request
+
+:::info
```
[the remark this answers is not in the document]
Comment (gosteenBASIS) [reply docs/advanced_topics/web-storage.md:41]: Added title "Cookie Expiration" as a title for this note.

File: docs/advanced_topics/web-storage.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+:::tip
+You can see current cookie, local storage and session storage key-value pairs in your browser's developer tools.
+:::
+
+### Summary of differences
+| Feature            | Cookies                                      | Session Storage                          | Local Storage                            |
+|--------------------|----------------------------------------------|------------------------------------------|------------------------------------------|
+| **Persistence**    | Configurable expiration date                 | Duration of the page session             | Persistent until explicitly deleted      |
+| **Storage Size**   | ~4KB per cookie                              | Around 5-10MB                            | Around 5-10MB                            |
```
[the remark this answers is not in the document]
Comment (gosteenBASIS) [reply docs/advanced_topics/web-storage.md:21]: Hyyan, what numbers would you expect here? These numbers look correct from what I can tell.
4,096 bytes is the minimum size of a cookie that all browsers should support:
https://en.wikipedia.org/wiki/HTTP_cookie#Implementation

Most Web browsers seem to support either 10 MB for Local and Session storage, and at least 5 MB:
https://en.wikipedia.org/wiki/Web_storage#Storage_size

File: docs/advanced_topics/web-storage.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+## Cookies
+Cookies are small pieces of data stored on the client side and sent to the server with each HTTP request. They are often used to remember user sessions, preferences, and authentication information.
+
+### Advantages:
+- Can store data across different domains
+- Support expiration dates
+
+### Disadvantages:
+- Small storage size (about 4KB)
```
[the remark this answers is not in the document]
Comment (gosteenBASIS) [reply docs/advanced_topics/web-storage.md]: The 4 KB limit seems to be widely accepted as a standard, and is a relevant consideration when using cookies. I would recommend we keep the specific size limit here.
https://datatracker.ietf.org/doc/html/rfc6265#section-6

File: docs/advanced_topics/web-storage.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+---
+import JavadocLink from '@site/src/components/DocsTools/JavadocLink';
+
+
+## Overview
+[Web storage](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API) is a fundamental concept in web development that allows websites to store data on the client side. This enables web applications to save state, preferences, and other information locally on the user's browser. Web storage provides a way to persist data across page reloads and browser sessions, reducing the need for repeated server requests and enabling offline capabilities.
+
+webforJ supports three mechanisms for storing client data: [**Cookies**](#cookies), [**Session Storage**](#session-storage), and [**Local Storage**](#local-storage).
+
+:::tip
```
Comment (hyyan) [inline docs/advanced_topics/web-storage.md:13]: Give this tip a title please

File: docs/advanced_topics/web-storage.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hyyan) [reply docs/advanced_topics/web-storage.md:21]: Thanks for sharing the source. Please include the links in the document

File: docs/advanced_topics/web-storage.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hyyan) [reply docs/advanced_topics/web-storage.md]: OK. That's fine. Keep it after including the link for the sizes in the table above.

**Review by hyyan — APPROVED** [review]

**Comment by hyyan** [comment]
@gosteenBASIS Looks Good. Thank you!

**Pull request merged**